.github/workflows/docs.yml

```diff
@@ -0,0 +1,31 @@
+---
+name: s2n-tls Documentation
+on: [push, pull_request]
+
+jobs:
+  generate-doxygen:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+      - name: Create Documentation
+        run: |
+            .github/s2n_doxygen.sh
+      - name: Upload documentation to action
+        # Upload the doxygen artifacts on pull requests to help reviewers easily
```

Comment:
I'm out of the loop, why would we be running this per pull request versus per merge? Doxygen is for our users, I'm not sure how helpful it is for a PR reviewer.

Reply:
In case someone is modifying the doxygen or the doxygen comments you can just download the zip instead of pulling in the PR and creating the doxygen.

After adding @dougch's suggestion this is less useful so I can just remove it